### PR TITLE
pool_test: simplify by using libvmi.NewCirros and tidying newPoolFromVMI up

### PR DESCRIPTION
**Pull request by dankenigsberg**
The test code is slightly clearer due to this change, as arcane implementation details are encapsulated (Cirros's requirement of coud-init userData has nothing to do with VM pools) and the signature of newPoolFromVMI() is made simpler.

```release-note
NONE
```

### CODE REVIEW

File: tests/pool_test.go [hunk 1]
```diff
@@ -114,13 +114,14 @@ var _ = Describe("[sig-compute]VirtualMachinePool", func() {
 
 	newVirtualMachinePool := func() *poolv1.VirtualMachinePool {
 		By("Create a new VirtualMachinePool")
-		template := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-		return newVirtualMachinePoolWithTemplate(template, true)
+		running := true
+		return createVirtualMachinePool(tests.NewRandomPoolFromVMI(libvmi.NewCirros(), int32(0), running))
```
Comment (xpivarc) [inline tests/pool_test.go:118]: What does `int32(0)` means? Either we define variables for both parameters or none.

Comment (dankenigsberg) [reply tests/pool_test.go:118]: I see your point - my hatred to boolean args has blinded me from seeing `replicas`'s opacity.

I can add; and I may also use this opportunity to move `NewRandomPoolFromVMI` out of the utils.go "junkyard" and next to its only user (this module). What do you say?

Comment (xpivarc) [reply tests/pool_test.go:118]: I think that would be great!

**Review by EdDev — APPROVED** [review]
Thanks!

**Review by enp0s3 — APPROVED** [review]
/approve
/retest

**Pull request merged**